xamarin/Xamarin.Forms
# [macOS] Fix disposing of NavigationToolbarTracker

### Description of Change ###

When we clean the `NavigationToolbarTracker` we need to pass the Element (NavigationPage) to know if we should remove or not the NavigationBar, current code was just passing null, so if NavigationTracker was tracking other element it would be replaced.

### Issues Resolved ### 

- fixes #9536

### API Changes ###
 
 None

### Platforms Affected ### 

- macOS

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 

Not applicable

### Testing Procedure ###

Go to issue 9536, push modal, then close it, you should still see the NavigationBar

### PR Checklist ###
